modules/ConnectedRouter.js

```diff
@@ -0,0 +1,52 @@
+import React, { Component, PropTypes } from 'react'
+import BrowserHistory from 'react-history/BrowserHistory'
+import StaticRouter from 'react-router/StaticRouter'
+
+import { LOCATION_CHANGE } from './reducer'
+
+class ConnectedRouter extends Component {
+  static propTypes = {
+    store: PropTypes.object,
+    history: PropTypes.func,
+    basename: PropTypes.string,
+    keyLength: PropTypes.number
+  }
+
+  static contextTypes = {
+    store: PropTypes.object
+  }
+
+  static defaultProps = {
+    history: BrowserHistory
+  }
+
+  render() {
+    const { history:History, basename, keyLength, ...props } = this.props
+
+    return (
+      <History location={props.location} basename={basename} keyLength={keyLength}>
+        {({ history, action, location }) => {
+          const store = this.context.store || this.props.store
+
+          store.dispatch({
```

Comment:
This could use some cleanup, but I tested it and it works without producing any warnings

```javascript
class ConnectedRouter extends Component {
  static propTypes = {
    store: PropTypes.object,
    history: PropTypes.func,
    basename: PropTypes.string,
    keyLength: PropTypes.number
  }

  static contextTypes = {
    store: PropTypes.object
  }

  static defaultProps = {
    history: BrowserHistory
  }

  render () {
    const { history: History, basename, keyLength, ...props } = this.props
    const store = this.context.store || this.props.store
    return (
      <History location={props.location} basename={basename} keyLength={keyLength}>
        {({history, action, location}) => (
          <BaseRouter {...{history, action, location, basename, ...props, store}} />) }
      </History>
    )
  }
}

class BaseRouter extends Component {
  componentDidMount () {
    this.updateLocation(this.props.location)
  }

  componentWillReceiveProps (nextProps) {
    this.updateLocation(nextProps.location)
  }

  updateLocation = (location) => {
    this.props.store.dispatch({
      type: LOCATION_CHANGE,
      payload: { location }
    })
  }

  render () {
    const { history, action, location, basename } = this.props
    return (
      <StaticRouter
        action={action}
        location={location}
        basename={basename}
        onPush={history.push}
        onReplace={history.replace}
        blockTransitions={history.block}
        {...this.props}
      />)
  }
}
```